src/Foundatio/Utility/SharedOptions.cs

```diff
@@ -0,0 +1,28 @@
+using System;
+using Foundatio.Serializer;
+using Microsoft.Extensions.Logging;
+
+namespace Foundatio {
+    public class SharedOptions {
+        public ISerializer Serializer { get; set; }
+        public ILoggerFactory LoggerFactory { get; set; }
+    }
+
+    public interface ISharedOptionsBuilder : IOptionsBuilder {}
```

Comment:
I think the `ISharedOptionsBuilder` is not neccessary. Since we can define the extension methods for `IOptionsBuilder` instead.